src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -75,6 +89,116 @@ public void doPut(HttpServletRequest request, HttpServletResponse response) thro
     response.sendRedirect("/index.html");
   }
 
+  private void addEvent(HttpServletRequest request, HttpServletResponse response, UserService userService)
+      throws IOException, ServletException {
+    String title = Objects.toString(request.getParameter("title"), "");
+    String description = Objects.toString(request.getParameter("description"), "");
+    String duration_parameter = request.getParameter("duration");
+    Long duration = 0L;
+    if (duration_parameter != null && !duration_parameter.isEmpty()) {
+      try {
+        duration = Long.parseLong(duration_parameter);
+      } catch (Exception e) {
+        System.err.println(e + duration_parameter);
+      }
+    }
+
+    String location = Objects.toString(request.getParameter("location"), "");
+
+    Map<String, String> fields = new HashMap<String, String>();
+    if (request.getParameterValues("fields") != null) {
+      for (String field : request.getParameterValues("fields")) {
+        fields.put(field, request.getParameter(field));
+      }
+    }
+
+    List<String> tags = new ArrayList<String>();
+    if (request.getParameterValues("tags") != null) {
+      tags = Arrays.asList(request.getParameterValues("tags"));
+    }
+
+    String current_user_id = userService.getCurrentUser().getUserId();
+    List<String> participants_ids = new ArrayList<String>();
+    if (request.getParameterValues("people") != null) {
+      participants_ids = Arrays.asList(request.getParameterValues("people"))
+                               .stream().map(person -> UserStorage.getIDbyUsername(person)).collect(Collectors.toList());
+    }
+
+    String startDateAsString = request.getParameter("start-date");
+    String startTimeAsString = request.getParameter("start-time");
+    String idFromGCalendarEvent = null;
+    if (startDateAsString != null && startTimeAsString != null && !startDateAsString.isEmpty() 
+         && !startTimeAsString.isEmpty()){// event has time set
+      com.google.api.services.calendar.model.Event newEvent = createGCalendarEvent(startTimeAsString, startDateAsString,
+          duration, location, description, title, fields);
+      if (newEvent == null){
+        response.setContentType("text/html");
+        response.getWriter().print("<script>alert(\"Please login first.\")</script>");
+        RequestDispatcher dispatcher = request.getRequestDispatcher("/add_event.html");
+        dispatcher.include(request, response);
+        return ;
+      }
+      idFromGCalendarEvent = newEvent.getId();
+    }
+    String idForDataStoreEvent = null;
+    if (idFromGCalendarEvent == null)
+      idForDataStoreEvent = UUID.randomUUID().toString();
+    else idForDataStoreEvent = idFromGCalendarEvent;
```

Comment:
Not sure whether it is a good idea to use google calendar id for datastore id, what happens on a conflict? For those events that don't have a calendar id yet, how will you store the Google calendar id then?